publisher.go

```diff
@@ -134,7 +137,7 @@ func (d *publisher) buildArgs(ctx context.Context, msgs []*MessageOutgoing) []an
 		for _, injector := range d.cfg.metaInjectors {
 			injector(ctx, msg.Metadata)
 		}
-		args = append(args, msg.Payload, msg.Metadata)
+		args = append(args, msg.Payload, msg.Metadata, msg.Delay)
```

Comment:
No idea how to comment unchanged row on GitHub, but it's about the L135
```
args := make([]any, 0, len(msgs)*3)
```

It preallocates the underlying array. Now the `append` must grow the array.

Reply:
Agree, I did not realised that and it works because you have a typo/bug :)
You should use 
```go
args = []any{msg.Payload, msg.Metadata, msg.Delay}
```
instead
```go
args = append(args, msg.Payload, msg.Metadata, msg.Delay)
```
with append you appending new values at the end of prealocated array, so final length of array will be 6 and benefit of prealocation is lost.

Reply:
Sorry, I reply before I think 😃 You are defining Length and Capacity

Personaly I would go with something like this. In this case the code returns error in case of my omit (*2 - *3) which is better then unexpected number of arguments.
```go
        args := make([]any, 0, len(msgs)*3)
	for i, msg := range msgs {
		for _, injector := range d.cfg.metaInjectors {
			injector(ctx, msg.Metadata)
		}
		if len(args)+3 >= i*3+3 {
			return errors.New("args slice is too short")
		}
		args[i*3] = msg.Payload
		args[i*3+1] = msg.Metadata
		args[i*3+2] = msg.Delay
	}
```
But mine is possibly slower because of len validation, and invalid cap can happend only by mistake in code. So I would say your approach is better + unit test :)